File: 11기 코딩테스트_230308/연습2/MostAppearInteger/MostAppearInteger_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.supereasy.MostAppearInteger.MostAppearInteger import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        s = 1559

        # when
        result = solution(s)

        # then
        self.assertEqual(5, result)

    def test_2(self):
        # given
        s = 19910326

        # when
        result = solution(s)

        # then
        self.assertEqual(1, result)

    def test_3(self):
        # given
        s = 98765432

        # when
        result = solution(s)

        # then
        self.assertEqual(2, result)

    def test_4(self):
        # given
        s = 10292938

        # when
        result = solution(s)

        # then
        self.assertEqual(2, result)

    def test_5(self):
        # given
        s = 1

        # when
        result = solution(s)

        # then
        self.assertEqual(1, result)

    def test_6(self):
        # given
        s = 9

        # when
        result = solution(s)

        # then
        self.assertEqual(9, result)


if __name__ == '__main__':
    unittest.main()

File: 11기 코딩테스트_230308/연습2/MostAppearInteger/MostAppearInteger_test.js
import assert from 'assert'
import solution from './MostAppearInteger.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const s = 1559

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(5, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const s = 19910326

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(1, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const s = 98765432

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(2, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const s = 10292938

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(2, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const s = 1

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(1, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const s = 9

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(9, result)
    done();
  });
})
